src/configFile.ts

```diff
@@ -0,0 +1,77 @@
+import { parse } from "toml";
+import { readFile } from "fs";
+import findUp from "find-up";
+interface foremanConfig {
+  tools: {
+    [tool_name: string]: foremanTool;
+  };
+}
+
+interface foremanTool {
+  source?: string;
+  github?: string;
+  gitlab?: string;
+  version: string;
+}
+
+const MANIFEST = "foreman.toml";
+
+function checkSameOrgToolSpecs(
+  manifestContent: foremanConfig,
+  org: string
+): boolean {
+  const tools = manifestContent.tools;
+  if (tools == null) {
+    throw new Error("Tools section in Foreman config not found");
+  }
+
+  for (const tool_name in tools) {
+    const tool_spec = tools[tool_name];
+    let source = tool_spec["source"];
+    if (source == null) {
+      source = tool_spec["github"];
+    }
+    if (source == null) {
+      continue;
+    }
+
+    const source_array = source.split("/");
+    const tool_org = source_array[0];
+
+    if (tool_org == null) {
+      throw new Error(
+        `Org not found in tool spec definition for: ${tool_name}. Foreman config is likely defined incorrectly.`
+      );
+    }
+
+    if (tool_org != org) {
+      return false;
+    }
+  }
+  return true;
+}
+
+async function checkSameOrgInConfig(org: string): Promise<void> {
+  const manifestPath = await findUp(MANIFEST);
+  if (manifestPath == null) {
+    throw new Error("setup-foreman could not find Foreman config file");
+  }
+
+  await readFile(manifestPath, "utf8", (err, data) => {
+    if (err) {
+      throw new Error("setup-foreman Could not read Foreman config file");
```

Comment:
```suggestion
      debug(err)
      throw new Error("setup-foreman Could not read Foreman config file");
```
it is generally a good idea to log error, to make troubleshooting easier